File: src/main/java/org/hwabeag/playerfly/schedules/FlyCheckTask.java
package org.hwabeag.playerfly.schedules;

import java.util.Iterator;
import java.util.Objects;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.GameMode;
import org.bukkit.World;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.hwabeag.playerfly.config.ConfigManager;
import org.hwabeag.playerfly.inventorys.FlySettingGUI;

public class FlyCheckTask implements Runnable {
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");
    String Prefix = ChatColor.translateAlternateColorCodes('&', (String)Objects.requireNonNull(PlayerFlyConfig.getString("playerfly.prefix")));

    public void run() {
        for (Player player : Bukkit.getOnlinePlayers()) {
            String name = player.getName();
            if (Objects.equals(PlayerFlyConfig.getString("플라이." + name + ".관리"), "온")) {
                World world = player.getWorld();
                String worldname = world.getWorldFolder().getName();
                for (String key : Objects.requireNonNull(PlayerFlyConfig.getConfigurationSection("playerfly.noworld")).getKeys(false)) {
                    if (Objects.equals(key, worldname)) {
                        if (player.getGameMode() == GameMode.SURVIVAL) {
                            if (player.getAllowFlight()) {
                                player.setAllowFlight(false);
                            }
                            player.setFlying(false);
                        }
                        PlayerFlyConfig.set("플라이." + name + ".관리", "오프");
                        ConfigManager.saveConfigs();
                        player.sendMessage(Prefix + " 플라이 사용이 금지된 월드입니다.");
                    }
                }
            } else if (player.getGameMode() == GameMode.SURVIVAL) {
                if (player.getAllowFlight()) {
                    player.setAllowFlight(false);
                }
                player.setFlying(false);
            }
        }
    }
}

File: src/main/kotlin/org/hwabeag/playerfly/schedules/FlyCheckTask.kt
package org.hwabeag.playerfly.schedules

import org.bukkit.Bukkit
import org.bukkit.GameMode
import org.bukkit.scheduler.BukkitRunnable
import org.hwabeag.playerfly.PlayerFlyPlugin

class FlyCheckTask(private val plugin: PlayerFlyPlugin) : BukkitRunnable() {
    override fun run() {
        for (player in Bukkit.getOnlinePlayers()) {
            val data = plugin.databaseManager.getData(player.uniqueId, player.name)
            if (!data.enabled) {
                continue
            }

            if (!plugin.isManagedFlight(player.uniqueId)) {
                plugin.markManagedFlight(player.uniqueId)
            }

            if (plugin.isBlockedWorld(player.world.name)) {
                if (player.gameMode == GameMode.SURVIVAL && plugin.isManagedFlight(player.uniqueId)) {
                    player.allowFlight = false
                    player.isFlying = false
                }
                plugin.databaseManager.setEnabled(player.uniqueId, player.name, false)
                plugin.unmarkManagedFlight(player.uniqueId)
                player.sendMessage("${plugin.prefix()} 이 월드에서는 플라이를 사용할 수 없습니다.")
            }
        }
    }
}
